### DISCUSSION TITLE
Generating more kubeconfigs for more users?

### DISCUSSION THREAD

**Discussion by echo-devnull** [rancher/rke2#3201] [General]
Hi All,

How could one generate more kubeconfigs with different CN's so I can create one for CICD, one that has limits to only one namespace, etc.

When using custom certificates I could create that manually, but in RKE2's case the certs are auto-generated.

What am I missing ?   ;-)

**Comment by brandond** [comment]
The certs are available on disk on any of the server nodes; if you know how to do it using custom certificates you can follow the same process using the certificates generated by RKE2.

Alternately, you could use Rancher, which has a UI for RBAC and user management, including generation of kubeconfigs for each user.

**Reply by echo-devnull** [reply]
This is just a one-node private "cluster" so I can try out different tactics (gitops, etc).  Learning ;-)
I know Rancher, and it's really cool for managing multiple clusters, but not really for me in this case. I want to learn the manual stuff first .

I'm trying out: https://kubernetes.io/docs/reference/access-authn-authz/certificate-signing-requests/

Thanks!
